clap_generate/src/generators/shells/zsh.rs

```diff
@@ -333,6 +333,37 @@ fn get_args_of(p: &App) -> String {
     ret.join("\n")
 }
 
+// Uses either `possible_vals` or `value_hint` to give hints about possible argument values
+fn value_completion(arg: &Arg) -> String {
+    if let Some(values) = &arg.possible_vals {
+        format!(
+            "({})",
+            values
+                .iter()
+                .map(|&v| escape_value(v))
+                .collect::<Vec<_>>()
+                .join(" ")
+        )
+    } else {
+        match arg.value_hint {
+            ValueHint::AnyPath => "_files",
+            ValueHint::FilePath => "_files",
```

Comment:
Are `AnyPath` and `FilePath` any different?

Reply:
Not for zsh, but to keep it flexible I imagine other shells might implement them differently.